docs/proposals/implement-api-that-imports-prometheus-blocks-into-cortex.md

```diff
@@ -0,0 +1,35 @@
+---
+title: "Implement API that imports Prometheus blocks into Cortex"
+linkTitle:
+weight:
+slug:
+---
+
+- Author: [Siddharth Asthana](https://github.com/edith007)
+- Date: March 2023
+- Status: draft
+
+## Problem statement
+
+There might be a lot of cases where the users might already be using Prometheus and later decides to switch to Cortex. Cortex will take care of the metrics which will be generated after the switch, but at present there is no way to backfill the old Prometheus TSDB blocks to Cortex. This proposal is to add the ability in cortex to fetch the Prometheus TSDB blocks and backfill the data.
+
+## Detailed Solution
+
+We can create a new service called `Importer` which will be responsible for bringing in the old Prometheus TSDB blocks to Cortex. 
```

Comment:
Do we need a new service for this API, or can we use existing service for it?

Reply:
@alanprot and I initially discussed that, and we thought that having an independent component would be nice 👍🏻

WDYT @yeya24 ??

Reply:
Umm, can you elaborate more why it is better than reusing existing components if we could?
Let's document this in our design doc, and elaborates the pros/cons of having a new service vs reusing existing services...

Reply:
I think having a different component is fair for this use case as we dont wanna this to interfere with any service on the hot path? (distributors and ingesters?). The only existing component that i see that may fit this is the purger (the one that is responsible to purge data from S3 upon tenant deletion), but adding this API in the purger seems weird (unless we rename it).

WDYT? @yeya24

Reply:
Sounds good to me 👍